handlers.go

```diff
@@ -262,30 +262,26 @@ func (dht *IpfsDHT) handleFindPeer(ctx context.Context, from peer.ID, pmes *pb.M
 
 	// if looking for self... special case where we send it on CloserPeers.
 	targetPid := peer.ID(pmes.GetKey())
-	if targetPid == dht.self {
-		closest = []peer.ID{dht.self}
-	} else {
-		closest = dht.betterPeersToQuery(pmes, from, dht.bucketSize)
-
-		// Never tell a peer about itself.
-		if targetPid != from {
-			// Add the target peer to the set of closest peers if
-			// not already present in our routing table.
-			//
-			// Later, when we lookup known addresses for all peers
-			// in this set, we'll prune this peer if we don't
-			// _actually_ know where it is.
-			found := false
-			for _, p := range closest {
-				if targetPid == p {
-					found = true
-					break
-				}
-			}
-			if !found {
-				closest = append(closest, targetPid)
+	closest = dht.betterPeersToQuery(pmes, from, dht.bucketSize)
```

Comment:
`dht.betterPeersToQuery` errors if the list contains self. 
https://github.com/libp2p/go-libp2p-kad-dht/blob/master/dht.go#L765-L769
```
		if clp == dht.self {
			logger.Error("BUG betterPeersToQuery: attempted to return self! this shouldn't happen...")
			return nil
		}
		// Dont send a peer back themselves
		if clp == from {
			continue
		}
```

It also handles the case that we don't inform the peer about itself.

Reply:
This snippet handles the output of [`routingTable.NearestPeers`](https://github.com/libp2p/go-libp2p-kad-dht/blob/1b5d0b69ec72752f651145f9f3a8aded499ab454/dht.go#L748), which never contains itself.

The peer adds it own peer id, only if its own peer id is the target of the `FIND_NODE` request at https://github.com/libp2p/go-libp2p-kad-dht/blob/7d6120fb0d6706c4ca76f5b3228747e0749670f5/handlers.go#L263-L285